### ISSUE TITLE
NPC errors/location unreachable

### ISSUE THREAD

**Issue by lvp11** [propstg/ambulance-mission#7]
Hey!

Sometimes the NPC's die before i can reach them, right after them spawn, exiting the mission for "NPC died" reason.

There's a unreachable location:
![image](https://user-images.githubusercontent.com/47780889/58812036-5fab7800-85f7-11e9-9533-0a14e1a0c66a.png)

Any fixes?

Thank you!

**Comment by propstg** [comment]
Hey, thanks for the heads up!

That location does seem a little tricky to get to, especially considering its distance from the hospital. I'll remove it from the config.

I have seen the instant NPC death a couple times, but it never happened frequently enough to determine the cause. I'll add something to hopefully prevent it.

**Comment by propstg** [comment]
Heya. I merged some changes. 

I removed that location from the config.

And, naturally, I wasn't able to get the insta-death to happen when I was trying to recreate it by spam-creating NPCs, so I wasn't able to verify that this fixed it... but I've merged a change that will _hopefully_ help prevent it. The NPCs should be invincible until the player gets close to them, so hopefully whatever is instantly killing them won't happen anymore.

Please let me know if you're still having issues! Thanks!

**Comment by lvp11** [comment]
Hey!

NPC's still dying randomly...
Any ideas?

Thanks!

**Comment by propstg** [comment]
Heya. 

Rats. Have you noticed if it's happening for a particular hospital?

I can take a look later today to try to track it down.

Thanks!

**Comment by lvp11** [comment]
Hey!

It's reported by my players, can't tell exactly where. 
I'll ask them and answer this more precisely, but i guess that's happening everywhere. 

Thanks.

**Comment by lvp11** [comment]
Hey!

It's reported by my players, can't tell exactly where. 
I'll ask them and answer this more precisely, but i guess that's happening everywhere. 

Thanks.

**Comment by propstg** [comment]
Heya.

I went through all the spawn points and spawned an NPC, dozens of times, and didn't have any instant deaths, so it doesn't seem to be an issue with any one spawn point in particular, so I bet you're right that it's happening everywhere.

I thought maybe it was the peds being deleted by the main thread when unloading and the other thread picking up that the ped is dead in the middle (and wrote a small test to confirm that this could be the cause), but then I realized that the other thread doesn't actually check the table that is being deleted :)

I still suspect it's a threading issue, but it wasn't immediately obvious what it could be, so I've pushed a change to not check for dead or dying peds when setting up a new level (or when unloading peds, but I doubt this is actually it).

Hopefully that helps!
